src/index.ts

```diff
@@ -8,8 +8,17 @@ import {
 } from '@codemirror/view';
 import { syntaxTree } from '@codemirror/language';
 import { Range } from '@codemirror/rangeset';
+import { namedColors } from './named-colors';
 
-const rgbCallExpRegex = /rgb\(\s*(\d{1,3})\s*,\s*(\d{1,3})\s*,\s*(\d{1,3})\s*\)/;
+enum ColorType {
+  rgb = "RGB",
+  hex = "HEX",
+  named = "NAMED",
+  hsl = "HSL"
+};
+
+const rgbCallExpRegex = /rgb\(\s*(\d{1,3}%?)\s*,\s*(\d{1,3}%?)\s*,\s*(\d{1,3}%?)\s*(,\s*0?\.\d+)?\)/;
+const hslCallExpRegex = /hsl\(\s*(\d{1,3})\s*,\s*(\d{1,3})%\s*,\s*(\d{1,3})%\s*(,\s*0?\.\d+)?\)/;
```

Comment:
I was reading the up on HSL and came across the fact that `hsl` and `rgb` in css support space separators instead of commas too!

Reply:
Yeah I didn't even try and handle the new syntax that supports spaces in places of commas - and also uses a '/' to separate the alpha. I figured the regex would get pretty funky if we wanted to really insist that if you start using command you must use commas for the rest (i.e. not just pop a '?' optional marker on each comma)